test/inductor/test_torchinductor.py

```diff
@@ -12145,18 +12145,35 @@ def f(x, mask):
 
         @requires_gpu()
         @parametrize("upcast_to_fp32", [False, True])
+        @config.patch("triton.use_block_ptr", True)
         def test_codegen_upcast_to_fp32(self, upcast_to_fp32):
             @torch.compile
-            def func(a, b):
-                return a * b
+            def func(a, b, c, d):
+                return a * b * c * d
 
-            inps = (torch.rand((32, 32), device=GPU_TYPE, dtype=torch.float16),) * 2
+            inps = (torch.rand((32, 32), device=GPU_TYPE, dtype=torch.float16),) * 4
             with config.patch("triton.codegen_upcast_to_fp32", upcast_to_fp32):
                 func_opt = torch._dynamo.optimize("inductor")(func)
                 code = run_and_get_triton_code(func_opt, *inps)
                 fp32_cast_in_code = "to(tl.float32)" in code
                 self.assertEqual(fp32_cast_in_code, upcast_to_fp32)
 
+        @requires_gpu()
+        @parametrize("load_upcast_to_fp32", [False, True])
+        @parametrize("input_dtype", [torch.float16, torch.bfloat16])
+        @config.patch("triton.use_block_ptr", True)
+        def test_dtype_aware_codegen(self, load_upcast_to_fp32, input_dtype):
```

Comment:
Nice test. Does this PR already affect the codegen? From the description, it sounded like that would be a follow up.

Reply:
We still keep the inductor flag `triton.codegen_upcast_to_fp32 = True` by default. Codegen will still see all CSEVariables in FP32 by default. I only updated the `sqrt` codegen so that I can put in an unit test to validate dtype-aware codegen works by switching off `triton.codegen_upcast_to_fp32` and feed in lower precision inputs.